LayoutTests/inspector/console/js-completions.html

```diff
@@ -52,6 +65,44 @@
         }
     });
 
+    suite.addTestCase({
+        name: "console.jsCompletions.completionOrdering",
+        description: "Test that the JavaScript completions are ordered correctly.",
+        async test() {
+            const outputCompletions = async (options) => {
+                InspectorTest.log(`Completions with options=${JSON.stringify(options)} and experimentalShowCaseSensitiveAutocomplete=${WI.settings.experimentalShowCaseSensitiveAutocomplete.value}:`);
+                InspectorTest.json(await generateJSCompletions(options));
+                InspectorTest.log("");
+            };
+
+            WI.settings.experimentalShowCaseSensitiveAutocomplete.value = false;
+            WI.javaScriptRuntimeCompletionProvider.clearCachedPropertyNames();
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: ""}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "a"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "A"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "_"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: ""}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "1"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"a"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"A"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"_"}));
+
+            WI.settings.experimentalShowCaseSensitiveAutocomplete.value = true;
+            WI.javaScriptRuntimeCompletionProvider.clearCachedPropertyNames();
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: ""}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "a"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "A"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds.", prefix: "_"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: ""}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "1"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"a"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"A"}));
+            await outputCompletions(({baseString: "objectWithMixedPropertyKinds[", prefix: "\"_"}));
+
+            WI.Setting.reset();
```

Comment:
NIT: we should really just be calling `WI.settings.experimentalShowCaseSensitiveAutocomplete.reset()` since that's the only thing we're modifying in this test, and `WI.Setting.reset()` also wont modify the cached `_value` stored in `WI.settings.experimentalShowCaseSensitiveAutocomplete`

Reply:
Ah, OK!